src/js/classes/NumberValidator.ts

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types';
+import {ensureValueIsNumeric} from '../helpers/developerInputValidators';
+import BaseValidator from './BaseValidator';
+import {Dependent, ErrorFunction, InputType, MutationFunction} from '../types';
+
+const numRegexp = /^-?\d+([.,]\d+)?$/s;
+const wholeNumRegexp = /^-?\d+$/s;
+
+const delocalize = (value: string | number): number => {
+    return parseFloat(String(value).replace(',', '.'));
+};
+
+/**
+ * A validator that allows you to validate numbers.
+ * @extends BaseValidator
+ */
+class NumberValidator extends BaseValidator<number|string> {
+    public defaultInputType: InputType = 'number';
+    protected propType: PropTypes.Requireable<any> = PropTypes.number;
+
+    protected _minNum?: number;
+    protected _maxNum?: number;
+
+    /**
+     * Validate a numeric input field
+     * @param wholeNumber - Set to true to only allow whole numbers
+     * @param {String|Number} [defaultValue]
+     * @param {String} [defaultErrorMsg]
+     * @param {MutationFunction} [mutationFunc]
+     * @param {ErrorFunction} [onError]
+     * @param {Dependent} [dependent]
+     */
+    constructor(
+        wholeNumber = false,
+        defaultValue: string | number = '',
+        defaultErrorMsg?: string,
+        mutationFunc?: MutationFunction,
+        onError?: ErrorFunction,
+        dependent?: Dependent
+    ) {
+        super(defaultValue, defaultErrorMsg, mutationFunc, onError, dependent);
+
+        let regexpInUse: RegExp;
+        let errorMsg = defaultErrorMsg ?? '';
+        if (wholeNumber) {
+            if (defaultErrorMsg === undefined) {
+                errorMsg = 'This field must be a whole number';
+            }
+            regexpInUse = wholeNumRegexp;
+        } else {
+            if (defaultErrorMsg === undefined) {
+                 errorMsg = 'This field must be a number';
+            }
+            regexpInUse = numRegexp;
+        }
+
+        this.validateFuncs.push(
+            (value) => {
+                const success = value !== undefined && regexpInUse.test(String(value));
```

Comment:
All undefined values should already be removed by the first check done in the validation function.  
Is this check necessary?

Reply:
Funny that you ask about this, cause I stumbled over this, too. No, I don't think it is necessary. However, some validate functions (specifically the notFromDomain one of the CustomEmailValidator) assumed the value could be undefined since before my conversation and I was about to ask you about these cases later. Let's talk about this on the call :)